UPGRADE-GUIDE.md

```diff
@@ -0,0 +1,465 @@
+command line version of Ganache from your [package.json scripts](https://docs.npmjs.com/cli/v8/using-npm/scripts) or directly
+from the command line using `npx ganache` ([what is npx?](https://www.npmjs.com/package/npx)).
+
+To upgrade a local installation using `npm` run these two commands:
+
+```console
+$ npm uninstall ganache-cli ganache-core
+$ npm install --save-dev ganache
+```
+
+_For security reasons[^1] we do not recommend using yarn or pnpm as your package
+manager when installing security-critical applications like Ganache._
+
+You will now be able to run Ganache programmatically from within your JavaScript
+or TypeScript files.
+
+An example diff might look like:
+
+```diff
+diff --git a/index.js b/index.js
+index abcdefg..hijklmn 123456
+--- a/index.js
++++ b/index.js
+@@ -0,7 +0,7 @@
+ const Web3 = require("web3");
+ const debug = require("debug");
+ const helpers = require("./helpers");
+-const ganache = require("ganache-core");
++const ganache = require("ganache");
+ const provider = ganache.provider();
+ const web3 = new Web3(provider);
+
+```
+
+Or in other words:
+
+```javascript
+// const ganache = require("ganache-core"); // <- replace this line with
+const ganache = require("ganache");
+```
+
+If you `require` or `import` the `ganache-cli` package instead of ganache-core
+the replacement is exactly the same:
+
+```javascript
+// const ganache = require("ganache-cli"); // <- replace this line with
+const ganache = require("ganache");
+```
+
+In your package.json scripts:
+
+```json
+{
+  "scripts": {
+    "start-ganache": "ganache"
+  }
+}
+```
+
+_Note: we've aliased ganache-cli to ganache, so you can continue using the
+ganache-cli command in your npm scripts and in your terminal._
+
+_note: you may need to uninstall using yarn, or another node package manager,
+depending on how you originally installed ganache-cli or ganache-core._
+
+
+
+### Breaking changes
+
+#### Allowed transaction signature changes
+
+Previously, Ganache allowed transaction signatures to original from 3 chains
+ ids: `1`, `chainId`, and `networkId`. Ganache now permits only the `chainId`.
+
+#### `_chainId` and `_chainIdRpc` options removed
+
+If you relied on the EVM's `CHAINID` opcode and the RPC method for `eth_chainId`
+to return different values you'll need to fix your code so that it only uses one
+of these values.
+
+The `_chainId` and `_chainIdRpc` options were a workaround for a legacy bug. We
+have not carried over this bug into Ganache v7. Use the `chainId` option from
+now on.
+
+#### `vmErrorsOnRPCResponse` now defaults to `false`
+
+The `vmErrorsOnRPCResponse` option caused Ganache to return an error for
+transactions that error or revert during EVM execution.
+
+On the command line this option was called `noVmErrorsOnRPCResponse`, and the
+default was `false`. In programmatic use this option was `vmErrorsOnRPCResponse`
+and it defaulted to `true`.
+
+Ganache now _disables_ the `vmErrorsOnRPCResponse` functionality by default.
```

Comment:
Might be worth putting _why_ we're disabling by default

Reply:
But I don't want to and I'm tired.